app/services/browser_service.py

```
@@ -41,6 +41,17 @@ def check_path_permission(db: Session, current_user: User, resolved_path: Path)
     if "admin" in current_user.roles:
         return
 
+    try:
+        # Resolve the path to eliminate any '..' traversal or symlink escapes
+        actual_path = resolved_path.resolve()
+    except OSError as e:
+        # Fail securely if the path cannot be resolved
+        logger.warning(f"Could not resolve path {resolved_path} for permission check.")
```

Ruff (0.15.11)

[warning] Logging statement uses f-string

(G004)